### PR TITLE
Fetch policy constants

### PR DESCRIPTION
**Pull request by NeoPhi**
TODO:

- [x] If this PR is a new feature, reference an issue where a consensus about the design was reached (not necessary for small changes)
- [x] Make sure all of the significant new logic is covered by tests
- [x] Rebase your changes on master so that they can be merged easily
- [x] Make sure all tests and linter rules pass
- [x] Update CHANGELOG.md with your change
- [x] Add your name and email to the AUTHORS file (optional)
- [ ] If this was a change that affects the external API, update the docs and post a link to the PR in the discussion

What to confirm the constant naming and approach for exporting in a namespace before updating any documentation.

### CODE REVIEW

File: src/core/ObservableQuery.ts [hunk 1]
```diff
@@ -453,15 +459,7 @@ export class ObservableQuery<T> extends Observable<ApolloQueryResult<T>> {
...
-      if (this.options.fetchPolicy === 'cache-first' || (this.options.fetchPolicy === 'cache-only')) {
-        throw new Error('Queries that specify the cache-first and cache-only fetchPolicies cannot also be polling queries.');
-      }
-
-      this.isCurrentlyPolling = true;
-      this.scheduler.startPollingQuery<T>(
-        this.options,
-        this.queryId,
-      );
+      this.startPolling(this.options.pollInterval);
```
Comment (NeoPhi) [inline src/core/ObservableQuery.ts]: Seemed a case of repeated logic so I opted to call the existing method instead.

File: src/core/fetchPolicy.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
+/**
+ * fetchPolicy determines where the client may return a result from.
+ */
+export type FetchPolicy = 'cache-first' | 'cache-and-network' | 'network-only' | 'cache-only';
```
Comment (helfer) [inline src/core/fetchPolicy.ts]: Constants are only useful for Javascript users. In Typescript they're not necessary at all. If we do use constants everywhere, this is the one line we definitely shouldn't skip!

File: src/core/fetchPolicy.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+/**
+ * return result from cache and only fetch from network if cached result is not available.
+ */
+export const CACHE_FIRST: FetchPolicy = 'cache-first';
+
+/**
+ * returns result from cache first (if it exists), then return network result once it's available.
```
Comment (helfer) [inline src/core/fetchPolicy.ts]: extra 's' in `returns`.

File: src/core/fetchPolicy.ts (hunk 2 above)
Comment (NeoPhi) [inline src/core/fetchPolicy.ts]: Wasn't sure of a way in TypeScript to define a type using constants. Instead I defined the constants to be of the FetchPolicy type.

**Comment by stubailo** [comment]
I'm wondering what the desired benefit of constants is in this case over strings - is it because there's a worry we might later need to change the underlying values of these constants? If we are going to keep the same string values forever, it seems like we don't get a lot of value out of wrapping them in constants.

File: src/core/fetchPolicy.ts (hunk 2 above)
Comment (stubailo) [inline src/core/fetchPolicy.ts]: This sounds like a job for a TypeScript Enum - I think you can have TS compile that into a dictionary with values.

**Comment by NeoPhi** [comment]
I couldn't find a way to do this cleanly in TypeScript with easy support for vanilla JS.

**Pull request closed without merging**